modeler.py

```diff
@@ -0,0 +1,59 @@
+from abc import ABCMeta, abstractmethod
+
+class Dynamics(metaclass=ABCMeta):
+    """Interface to dynamics model."""
+
+    @abstractmethod
+    def __call__(self, state, action):
+        """Simulate a world transition.
+
+        Args:
+          state (object): Current world state to start from.
+          action (object): Action to take in current world state.
+        
+        Returns:
+          (object): Next state after taking the given action in the given state.
+        """
+        pass
+
+class PerfectDynamics(Dynamics, metaclass=ABCMeta):
+    """Interface to perfect information dynamics model."""
+
+    @abstractmethod
+    def __call__(self, state, action):
+        """Simulate a world transition.
+
+        Args:
+          state (object): Current world state to start from.
+          action (object): Action to take in current world state.
+        
+        Returns:
+          (object): Next state.
+          (float): Reward or final state value.
+          (bool): Is terminal.
+        """
+        pass
+
+
+class Modeler(metaclass=ABCMeta):
+    """Modeler interface to world model learner."""
+
+    @abstractmethod
+    def get_dynamics():
+        """Passes world dynamics used to simulate transitions.
+        
+        Return:
+          (Dynamics): World dynamics.
+        """
+        pass
+
+    @abstractmethod
+    def report_step(self, transition):
```

Comment:
I think but this method should be moved to separated interface named Callback

Reply:
Will you do this?

Reply:
yes, I can